### ISSUE TITLE
[ENH] Codification of `keptn` Code

### ISSUE THREAD

**Issue by afzal442** [keptn/keptn#7312]
`keptn` is having low score on [goreportcard](https://goreportcard.com/report/github.com/keptn/keptn#golint). As per this, it is needed to enhance the code to make it as much as lint error free code. 

**Environment**

* Client OS (e.g., Linux, macOS, Windows):
* Keptn Version (`keptn version`):
* Kubernetes Cloud Provider (e.g., GKE, AKS):
* Kubernetes version (`kubectl version`):

**Comment by jainammm** [comment]
can you assign this issue to me? @afzal442... thanks

**Comment by jainammm** [comment]
Hey @afzal442, the [goreportcard](https://goreportcard.com/report/github.com/keptn/keptn) was last updated 10 months ago... On refreshing it's giving an error of `no go files found` probably because there aren't any `go` files in the root folder 🤷‍♂️

**Comment by afzal442** [comment]
hi Jainam, thanks much! 
> to fix golint, we'll have to either address all comments or configure golint to exclude certain things like uncommented exported functions

Sounds good. We can do that either.
> On refreshing it's giving an error of no go files found

Yes, you are right. It's issue with that. Let's wait for this fix. 😄

**Comment by afzal442** [comment]
>  suggest we add a static lint check `golint` to the pipeline to prevent PRs which don't adhere to golang convention from being merged

This is pretty good idea to prevent those warnings. 💯  Let's see what maintainers have a say on it.

**Comment by jainammm** [comment]
Hey @afzal442, regardless of old report on `goreportcard` website we can make changes for this. I've linked a draft PR for it too with fix for formatting :) 

> Sounds good. We can do that either.

Okay, cool. I'll make a call on that and let you know. 

> This is pretty good idea to prevent those warnings. 💯 Let's see what maintainers have a say on it.

Yes sure 👍 

Thanks.

**Comment by jainammm** [comment]
For `ineffassign`… most of the cases are where `err` variable is assigned by not handled… but i believe it’s there for readability purpose… so replacing it with `_` won’t be a better approach 🤷‍♂️ 

so i think we should ignore those errors... what are your thoughts on that?

**Comment by jainammm** [comment]
As `goreportcard` website is failing... To get the latest list of warnings I ran this command `golint ./...`... It gave more than 1000 warnings... But after removing the exported comment not present warnings it boils down to around 268 warnings... Those are mainly package and function name changes only... I'll address them individually in the linked PR.

PFA the list of such warnings. 
[golint_warnings.txt](https://github.com/keptn/keptn/files/8408235/golint_warnings.txt)

**Comment by afzal442** [comment]
Thanks Jainam, yup. You are in the right direction, good job. 🛩️

**Comment by jainammm** [comment]
Hey, 

I've addressed around 90 comments and updated in the PR too.. I keep committing based on some file changes.

Though I change the stuct names and other names based on warning from `golint` and Golang coding conventions... I have one doubt... Some other repositories could be importing from files in this repo... Wouldn't that affect those imports? I had this doubts because I addressed lots of comments asking to change naming by removing `_`... But I was surprised to see that the occurrences of them in `keptn` repo was only once... Examples of such files are `helm-service/pkg/apis/networking/istio/v1alpha3/virtual_service.go` and `helm-service/pkg/apis/networking/istio/v1alpha3/destination_rule.go`
Or are those files deprecated rn? 


PS: Following comments are still pending. Will keep on addressing them as I get time to work on them. 
[golint_warnings.txt](https://github.com/keptn/keptn/files/8457750/golint_warnings.txt)

Thanks!

cc: @afzal442
